### DISCUSSION TITLE
Pipelines framework vs WebUI tools & functions

### DISCUSSION THREAD

**Discussion by nomppy** [open-webui/pipelines#180] [General]
I'm pretty new to open-webui and am interested in using the tools/pipelines feature a lot, and given the similarities between pipelines (this repo) and native calling implementation on the webui, I'm wondering how people are differentiating their usecase between these two systems.

From what I these are the differences:

**Pipelines framework**: 
- Separate load from ui
- Easier to install arbitrary dependencies 
- Compatible with non Open-webui frameworks (but only pipes, not filters)

**Functions & tools**:
- Access to [community](https://openwebui.com/) code
- Can enable/disable toolkits in UI
- (Recently) User feedback with `__event_emitter__` (https://github.com/open-webui/open-webui/releases/tag/v0.3.9)

Between the community and recent updates to Functions & tools, and what @tjbck said in [this PR](https://github.com/open-webui/open-webui/pull/798#issuecomment-2143593842):
> Our plan is to push https://github.com/open-webui/pipelines as THE plugin system for Open WebUI.

I'm wondering what the direction is going to be in the future? Is one system going to be a strict subset of the other? 

Related: https://github.com/open-webui/open-webui/discussions/3134

**Comment by justinh-rahb** [comment]
We love all of our children equally! 😅

Your summary of the lay of the land as of now is mostly correct. The plan is to keep supporting both, and try to keep as much feature parity as we can, within the constraints the respective platforms allow.

**Reply by sir3mat** [reply]
__event_emitter__ are compatible with pipelines?
